js/ui/components/clubhouse.js

```diff
@@ -473,6 +473,9 @@ class ClubhouseNotificationBanner extends MessageTray.NotificationBanner {
 
         button.set_x_expand(false);
         super.addButton(button, callback);
+
+        // @todo: This is a workaround for the missing CSS selector, :nth-child
+        button.add_style_class_name('child-' + this._buttonBox.get_n_children());
```

Comment:
I've been looking into the shell code to view why the nth-child is not implemented and the code is not easy to read, I don't understand yet the whole StWidget and css parsing, so it's not easy.

:+1: This is exactly what comes to my mind when I see that it was not easy to implement the nth-child selector. It's an elegant solution in our custom code that provide the required functionality for the css.

Reply:
It would take a considerable effort to implement nth-child in St. The biggest challenge would be switching to use a more efficient data structure, like a GSequence, instead of a GList.